core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/newresource/AbstractNewResourceAction.java

```diff
@@ -244,30 +244,8 @@ protected String getDefaultContent() {
     /** Returns parent for creating new item or {@code null} if resource can not be created. */
     @Nullable
     protected ResourceBasedNode<?> getResourceBasedNode() {
-        Selection<?> selection = projectExplorer.getSelection();
-
-        //we should be sure that user selected single element to work with it
-        if (selection == null || selection.isEmpty()) {
-            return null;
-        }
-
-        Object o = selection.getHeadElement();
-
-        if (o instanceof ResourceBasedNode<?>) {
-            ResourceBasedNode<?> node = (ResourceBasedNode<?>)o;
-            //it may be file node, so we should take parent node
-            if (node.isLeaf() && isResourceAndStorableNode(node.getParent())) {
-                return (ResourceBasedNode<?>)node.getParent();
-            }
-
-            return isResourceAndStorableNode(node) ? node : null;
-        }
-
-        return null;
-    }
-
-    protected boolean isResourceAndStorableNode(@Nullable Node node) {
-        return node != null && node instanceof ResourceBasedNode<?> && node instanceof HasStorablePath;
+        BasicUploadPresenter basicUploadPresenter = new BasicUploadPresenter(projectExplorer);
```

Comment:
Not, good use create presenter with constructor try please use injection instead. Thanks